includes/class-edd-discount.php

```diff
@@ -1123,32 +1163,26 @@ public function update( $args = array() ) {
 		}
 
 		if ( isset( $args['excluded_products'] ) ) {
-			if ( is_array( $args['excluded_products'] ) ) {
-
-				// Reset meta
-				$this->delete_meta( 'excluded_product' );
+			// Reset meta
+			edd_delete_adjustment_meta( $this->id, 'excluded_product' );
 
+			if ( is_array( $args['excluded_products'] ) ) {
 				// Now add each newly excluded product
 				foreach ( $args['excluded_products'] as $product ) {
-					$this->add_meta( 'excluded_product', absint( $product ) );
+					edd_add_adjustment_meta( $this->id, 'excluded_product', absint( $product ) );
 				}
-			} else {
-				$this->delete_meta( 'excluded_product' );
 			}
 		}
 
 		if ( isset( $args['product_reqs'] ) ) {
-			if ( is_array( $args['product_reqs'] ) ) {
-
-				// Reset meta
-				$this->delete_meta( 'product_requirement' );
+			// Reset meta
+			edd_delete_adjustment_meta( $this->id, 'product_reqs' );
```

Comment:
I think this is not quite right--the $args are `product_reqs`, but the database key is `product_requirement`?
```suggestion
			edd_delete_adjustment_meta( $this->id, 'product_requirement' );
```